persists/backend_server/server.bal

```diff
@@ -0,0 +1,65 @@
+import ballerina/http;
+import ballerina/persist;
+import ballerina/random;
+
+final Client orderDatabase = check new;
+
+@http:ServiceConfig {
+    cors: {
+        allowOrigins: ["*"]
+    }
+}
+service /sales on new http:Listener(9090) {
+    // Example: http://localhost:9090/sales/orders
+    isolated resource function get orders() returns Order[]|error {
+        return from Order entry in orderDatabase->/orders(targetType = Order)
+            select entry;
+    };
+
+    // Example: http://localhost:9090/sales/orders/HM-238
+    isolated resource function get orders/[string id]() returns Order|http:BadRequest {
+        Order|error orderEntry = orderDatabase->/orders/[id];
```

Comment:
Shall we return the error first? IMO it is more natural to return error first.

```ballerina
        Order|error orderEntry = orderDatabase->/orders/[id];
        if orderEntry is error {
            return <http:BadRequest>{
                body: {
                    message: string `Error while retrieving the order, ${orderEntry.message()}`
                }
            };
        }
        return orderEntry;
```

Reply:
I remember a discussion regarding putting the `return orderEntry` also within an else condition.

Reply:
We discussed to handle different `persist` error types inside the service.
That implementation is demonstrated in the `L47`.
I will do it for this resource function as well.

Reply:
so it will be updated as
```
    isolated resource function get orders/[string id]() returns Order|http:NotFound|http:InternalServerError {
        Order|persist:Error orderEntry = orderDatabase->/orders/[id];
        if orderEntry is persist:NotFoundError {
            return <http:NotFound>{body: {message: string `Order not found, id: ${id}`}};
        } else if orderEntry is persist:Error {
            return <http:InternalServerError>{
                body: {message: string `Error while retrieving the order ${orderEntry.message()}`}
            };
        } else {
            return orderEntry;
        }
    };
```